common/src/OpenSpaceNet.cpp

```diff
@@ -363,359 +400,282 @@ void OpenSpaceNet::initFilter()
-                    predictions = filterLabels(predictions, LabelFilterType::INCLUDE, includeLabels);
-                }
-
-                for(auto& prediction : predictions) {
-                    prediction.window.x += item.first.x;
-                    prediction.window.y += item.first.y;
-                    addFeature((PredictionPoly)prediction);
-                }
-            }
+    DG_CHECK(!args_.resampledSize || *args_.resampledSize <= modelSize_.width,
+             "Argument --resample-size (size: %d) does not fit within the model (width: %d).",
+             *args_.resampledSize, modelSize_.width)
 
-            if(pd_ && !args_.quiet) {
-                pd_->update(classifyCategory_, (float)++curBlockClass / numBlocks);
-            }
+    if (!args_.resampledSize) {
+        for (auto c : args_.windowSize) {
+            DG_CHECK(c <= modelSize_.width,
+                     "Argument --window-size contains a size that does not fit within the model (width: %d).",
+                      modelSize_.width)
         }
-    });
-
-    consumerFuture.wait();
-    if(pd_ && !args_.quiet) {
-        pd_->stop();
     }
 
-    image_->rethrowIfError();
+    Detector::Ptr modelNode;
+    if(metadata_->category() == "segmentation") {
+        initSegmentation(model);
+        modelNode = deepcore::classification::node::PolyDetector::create("Model");
+    } else {
+        modelNode = deepcore::classification::node::BoxDetector::create("Model");
+    }
 
-    skipLine();
+    modelNode->attr("model") = model;
+    modelNode->attr("confidence") = confidence;
+    return modelNode;
 }
 
-void OpenSpaceNet::processSerialPolys()
+void OpenSpaceNet::initSegmentation(Model::Ptr model)
 {
-    auto predictions = processSerial<PredictionPoly>();
-
-    if(args_.action != Action::LANDCOVER && args_.nms) {
-        skipLine();
-        OSN_LOG(info) << "Performing non-maximum suppression..." ;
-        predictions = polygonalNonMaxSuppression(predictions, args_.overlap / 100);
-    }
+    auto castModel = dynamic_cast<CaffeSegmentation*>(model.get());
+    DG_CHECK(castModel, "Unsupported model type: only Caffe segmentation models are currently supported");
 
-    OSN_LOG(info) << predictions.size() << " features detected.";
+    castModel->setRasterToPolygon(make_unique<RasterToPolygonDP>(args_.method, args_.epsilon, args_.minArea));
+}
 
-    for(const auto& prediction : predictions) {
-        bool skip = false;
-        for(const auto& point : prediction.poly.exteriorRing.points) {
-            if(point.x < bbox_.x || point.y < bbox_.y || point.x > bbox_.width || point.y > bbox_.height) {
-                skip = true;
-                break;
-            }
-        }
+dg::deepcore::imagery::node::SlidingWindow::Ptr OpenSpaceNet::initSlidingWindow()
+{
+    auto slidingWindow = dg::deepcore::imagery::node::SlidingWindow::create("SlidingWindow");
```

Comment:
Didn't you import the whole `dg::deepcore::imagery::node` namespace? If you did, there shouldn't be a need to fully qualify it here. Or is there ambiguity because whole namespaces are imported?

Reply:
For nodes that share names with their non-node counterparts, there is ambiguity. There is a similar issue with the detector nodes.